src/analysis/normed_space/inner_product.lean

```diff
@@ -733,25 +733,62 @@ include 𝕜
 lemma parallelogram_law_with_norm {x y : E} :
   ∥x + y∥ * ∥x + y∥ + ∥x - y∥ * ∥x - y∥ = 2 * (∥x∥ * ∥x∥ + ∥y∥ * ∥y∥) :=
 begin
-  simp only [(inner_self_eq_norm_square _).symm],
-  rw[←add_monoid_hom.map_add, parallelogram_law, two_mul, two_mul],
-  simp only [add_monoid_hom.map_add],
+  simp only [← inner_self_eq_norm_square],
+  rw[← re.map_add, parallelogram_law, two_mul, two_mul],
+  simp only [re.map_add],
 end
 omit 𝕜
 
 lemma parallelogram_law_with_norm_real {x y : F} :
   ∥x + y∥ * ∥x + y∥ + ∥x - y∥ * ∥x - y∥ = 2 * (∥x∥ * ∥x∥ + ∥y∥ * ∥y∥) :=
 by { have h := @parallelogram_law_with_norm ℝ F _ _ x y, simpa using h }
 
+/-- Polarization identity: The real part of the  inner product, in terms of the norm. -/
+lemma re_inner_eq_norm_add_mul_self_sub_norm_mul_self_sub_norm_mul_self_div_two (x y : E) :
+  re ⟪x, y⟫ = (∥x + y∥ * ∥x + y∥ - ∥x∥ * ∥x∥ - ∥y∥ * ∥y∥) / 2 :=
+by { rw norm_add_mul_self, ring }
+
+/-- Polarization identity: The real part of the  inner product, in terms of the norm. -/
+lemma re_inner_eq_norm_mul_self_add_norm_mul_self_sub_norm_sub_mul_self_div_two (x y : E) :
+  re ⟪x, y⟫ = (∥x∥ * ∥x∥ + ∥y∥ * ∥y∥ - ∥x - y∥ * ∥x - y∥) / 2 :=
+by { rw [norm_sub_mul_self], ring }
+
+/-- Polarization identity: The real part of the  inner product, in terms of the norm. -/
+lemma re_inner_eq_norm_add_mul_self_sub_norm_sub_mul_self_div_four (x y : E) :
+  re ⟪x, y⟫ = (∥x + y∥ * ∥x + y∥ - ∥x - y∥ * ∥x - y∥) / 4 :=
+by { rw [norm_add_mul_self, norm_sub_mul_self], ring }
+
+/-- Polarization identity: The imaginary part of the inner product, in terms of the norm. -/
+lemma im_inner_eq_norm_sub_I_smul_mul_self_sub_norm_add_I_smul_mul_self_div_four (x y : E) :
+  im ⟪x, y⟫ = (∥x - IK • y∥ * ∥x - IK • y∥ - ∥x + IK • y∥ * ∥x + IK • y∥) / 4 :=
+by { simp only [norm_add_mul_self, norm_sub_mul_self, inner_smul_right, I_mul_re], ring }
+
+/-- Polarization identity: The inner product, in terms of the norm. -/
+lemma inner_eq_sum_norm_sq_div_four (x y : E) :
+  ⟪x, y⟫ = (∥x + y∥ ^ 2 - ∥x - y∥ ^ 2 + (∥x - IK • y∥ ^ 2 - ∥x + IK • y∥ ^ 2) * IK) / 4 :=
+begin
+  rw [← re_add_im ⟪x, y⟫, re_inner_eq_norm_add_mul_self_sub_norm_sub_mul_self_div_four,
+    im_inner_eq_norm_sub_I_smul_mul_self_sub_norm_add_I_smul_mul_self_div_four],
+  push_cast,
+  simp only [pow_two, ← mul_div_right_comm, ← add_div]
+end
+
+/-- A linear isometry preserves the inner product. -/
+lemma linear_isometry.inner_map_map {E' : Type*} [inner_product_space 𝕜 E'] (f : E →ₗᵢ[𝕜] E')
+  (x y : E) : ⟪f x, f y⟫ = ⟪x, y⟫ :=
+by simp [inner_eq_sum_norm_sq_div_four, ← f.norm_map]
```

Comment:
Could you also provide a constructor for a `linear_isometry`, given a linear map which preserves the inner product?  I know it's only a couple of lines, but the argument will probably be needed more thanonce.